pkg/buildpod/generator.go

```diff
@@ -10,11 +10,16 @@ import (
 	"github.com/pkg/errors"
 	corev1 "k8s.io/api/core/v1"
 	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
 	k8sclient "k8s.io/client-go/kubernetes"
+	"knative.dev/pkg/apis/duck"
 
 	"github.com/pivotal/kpack/pkg/apis/build/v1alpha1"
+	"github.com/pivotal/kpack/pkg/apis/build/v1alpha2"
```

Comment:
Is it really the best approach to mix `build/v1alpha1` and `build/v1alpha2`? It can get very tricky which struct to take from which import and the errors can be very subtle.

Another approach would bump the  api as a whole and only use `build/v1alpha2`. Maybe even use a import like

```bash
buildapi "github.com/pivotal/kpack/pkg/apis/build/v1alpha2"
```

So the code didn't have to be changed all over if a version is bumped.